processing/src/main/java/io/druid/query/search/SearchBinaryFn.java

```diff
@@ -62,24 +63,49 @@ public SearchBinaryFn(
       return arg1;
     }
 
+    final int limit = gran instanceof AllGranularity ? this.limit : -1;
```

Comment:
I believe its for doing the exact count ? 
probably we can still have limit when we dont need count ?